### PR TITLE
Dash/election demo

### PR DESCRIPTION
**Pull request by jomar-honsi**
- PoC functionality
- Without React and Grid editing
- Responsiveness issues remain

### CODE REVIEW

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
It doesn't look good on mobile/ipad:
image

**Comment by sebastianbochan** [comment]
1. Add some padding-bottom space
Screenshot 2024-03-05 at 09 40 52

2. Images are not good, I suggest using square images and applying `border-radius` to achieve a circle. 
Screenshot 2024-03-05 at 09 40 41

3. Datagrid is cut at the right edge
Screenshot 2024-03-05 at 09 40 31

I suggest applying a mobile style for 992px as well. The squeezed map and the column chart do not look good on the tablet.

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

File: samples/dashboards/demo/election/demo.css [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+}
+
+.row {
+    display: flex;
+    flex-direction: row;
+}
+
+.cell {
+    margin: 10px;
+    min-width: 40%;
```
[the remark this answers is not in the document]
Comment (jomar-honsi) [reply samples/dashboards/demo/election/demo.css]: > Can we replace min-width with `flex: 1`? It is better to apply margin on `.cell > .highcharts-dashboards-component`
> 
> I suggest taking a look at the rest of demos, which are updated to the newest v2.

Done

**Comment by jomar-honsi** [comment]
> 1. Add some padding-bottom space
>    Screenshot 2024-03-05 at 09 40 52
> 2. Images are not good, I suggest using square images and applying `border-radius` to achieve a circle.
>    Screenshot 2024-03-05 at 09 40 41
> 3. Datagrid is cut at the right edge
>    Screenshot 2024-03-05 at 09 40 31
> 
> I suggest applying a mobile style for 992px as well. The squeezed map and the column chart do not look good on the tablet.

Issue 1 and 3 fixed. Images changed to use 1:1, still working on scaling issue.

File: samples/dashboards/demo/election/demo.css [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+    --color-hover-background: #5352be;
+    --color-hover-border: #e7e7ec;
+    --color-hover-text: #e7e7ec;
+}
+
+@media (prefers-color-scheme: dark) {
+    /* dark mode */
+    :root {
+        /* Democrat */
+        --highcharts-color-0: rgb(74, 131, 240);
```
[the remark this answers is not in the document]
Comment (jomar-honsi) [reply samples/dashboards/demo/election/demo.css]: Done

File: samples/dashboards/demo/election/demo.js [hunk 3]
```diff
@@ -0,0 +1,824 @@
+/* eslint-disable max-len */
```
[the remark this answers is not in the document]
Comment (jomar-honsi) [reply samples/dashboards/demo/election/demo.js:1]: Removed

File: samples/dashboards/demo/election/demo.css [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+
+#election-map .highcharts-tooltip-box {
+    fill: #9eb8f9;
+    fill-opacity: 1;
+    stroke: none;
+}
+
+/* Custom map tooltip */
+
+#election-map .map-tooltip caption {
```
[the remark this answers is not in the document]
Comment (jomar-honsi) [reply samples/dashboards/demo/election/demo.css:367]: To be discussed, awaiting feedback from @nasin

**Comment by jomar-honsi** [comment]
> It doesn't look good on mobile/ipad: image

Resolved (apart from image scaling issue).

File: samples/dashboards/demo/election/demo.html [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        <li><b>Highcharts Chart</b> shows historical national or state results.</li>
+        <li><b>Dashboards Datagrid</b> shows detailed results of all states. In the grid the electoral vote may be
+            changed.</li>
+    </ul>
+    <h2>Data sources</h2>
+    <p>The data used in this demo are provided by Wikipedia.com and the U.S. National Archives.</p>
+</div>
+
+<!-- XML data island: data for Custom HTML components-->
+<xml id="xml-data-island">
```
[the remark this answers is not in the document]
Comment (jomar-honsi) [reply samples/dashboards/demo/election/demo.html:46]: I'm unable to reproduce the scroll issue. Could you provide some more details?

[the remark this answers is not in the document]
Comment (karolkolodziej) [reply samples/dashboards/demo/election/demo.html:46]: see:
image

File: samples/dashboards/demo/election/demo.html [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+                <candidate>McCain</candidate>
+                <imgUrl>https://assets.highcharts.com/demos/samples/graphics/dashboards/election/mccain.jpg</imgUrl>
+            </rep>
+            <dem>
+                <candidate>Obama</candidate>
+                <imgUrl>https://assets.highcharts.com/demos/samples/graphics/dashboards/election/obama.jpg</imgUrl>
+            </dem>
+        </year>
+    </elections>
+</xml>
```
Comment (Vazonik) [inline samples/dashboards/demo/election/demo.html:171]: Missing trailing line.

File: samples/dashboards/demo/election/demo.css [hunk 7]
```diff
@@ -0,0 +1,571 @@
...
+/* stylelint-disable-next-line max-line-length */
+.highcharts-dark #election-map .highcharts-series-1 .highcharts-data-label text {
+    paint-order: stroke;
+    stroke-width: 2px;
+}
+
+/* Dashboards layout */
+
+/* LARGE */
+@media (max-width: 1200px) {
```
Comment (Vazonik) [inline samples/dashboards/demo/election/demo.css:491]: I wonder if it wouldn't be better to switch `@media` queries to `@container` in future demos. In this case, responsiveness would result from the width of the board, not the entire page. https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries

See others' input from the accounting demo: https://github.com/highcharts/highcharts/pull/20220#issuecomment-1958909467

And then mobile view in utils will also work.

**Review by Vazonik — COMMENTED** [review]
Looks promising! A few comments from me for now.

**Comment by sebastianbochan** [comment]
1. Spacings are not the same (mobile view)
Screenshot 2024-03-07 at 13 45 53

2. Missing padding below the text
Screenshot 2024-03-07 at 13 45 36

3. How about this kind of style avatars? I suggest adding some more space (around central chart), increase border radius in avatars and remove the border around them.
Screenshot 2024-03-07 at 13 44 52

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
Candidates are not visible on mobile:
image

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
With a wider screen, the result bar is not visible:
image

**Comment by jomar-honsi** [comment]
Fixed.

**Comment by jomar-honsi** [comment]
Good catch!

**Review by karolkolodziej — APPROVED** [review]
Please check the alerts. Other than that looks good!

**Review by sebastianbochan — APPROVED** [review]
Should we have bolded the first row when scrolling?

bold

Otherwise looks good 👍

**Comment by jomar-honsi** [comment]
> Should we have bolded the first row when scrolling?
> 
> bold [ bold ](https://private-user-images.githubusercontent.com/2115686/313642414-5f186c0e-0f6a-47db-96b9-f861b3751860.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3OTA3NTcsIm5iZiI6MTcxMDc5MDQ1NywicGF0aCI6Ii8yMTE1Njg2LzMxMzY0MjQxNC01ZjE4NmMwZS0wZjZhLTQ3ZGItOTZiOS1mODYxYjM3NTE4NjAuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMThUMTkzNDE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTQzN2E1OThjZjkyMzA1MWY5Mzc0MGEyOTU3YTM0M2U2NTI5YjM0ZDc2MmNhODY0OTY4OGM4NTdjMDY3YmEyMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.cuP84e2v20WJNrhhp4vN53RheP9MP17xi-bLfR-_lTc) [ ](https://private-user-images.githubusercontent.com/2115686/313642414-5f186c0e-0f6a-47db-96b9-f861b3751860.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3OTA3NTcsIm5iZiI6MTcxMDc5MDQ1NywicGF0aCI6Ii8yMTE1Njg2LzMxMzY0MjQxNC01ZjE4NmMwZS0wZjZhLTQ3ZGItOTZiOS1mODYxYjM3NTE4NjAuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMThUMTkzNDE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTQzN2E1OThjZjkyMzA1MWY5Mzc0MGEyOTU3YTM0M2U2NTI5YjM0ZDc2MmNhODY0OTY4OGM4NTdjMDY3YmEyMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.cuP84e2v20WJNrhhp4vN53RheP9MP17xi-bLfR-_lTc)
> 
> Otherwise looks good 👍

Hmm., that was not the intention. The first row should only be bold when it shows the U.S. federal results. Any idea how to achieve this?

**Comment by sebastianbochan** [comment]
@jomar-honsi you can use the data field.

I just committed it.

**Review by jomar-honsi — COMMENTED** [review]
Nice!

**Review by Vazonik — APPROVED** [review]
Great!

**Review by stitot — APPROVED** [review]

**Pull request merged**